### PR TITLE
Mask EUII logs

### PR DESCRIPTION
**Pull request by oldalton**
## Proposed changes

MSAL and broker provide logging callbacks that assist in diagnostics. By default the library will not return any messages with any user or organizational information. However, this might make diagnosing issues difficult.

The most sensitive information is EUII (things like upn, name, email etc). 
This PR introduces a new logger flag and macro that allows masking only EUII, while keeping EUPI and OII intact. 

When both piiEnabled is set to YES, and maskEUII is set to YES, MSAL logs will still include OII (organization identifiable information), and EUPI (end user pseudonymous identifiers), but MSAL will try to exclude and/or mask any EUII (end user identifiable information) like UPN, username, email from its logs.

This flag has no effect when piiEnabled is set to NO.
Default value is NO.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

## Additional information

### CODE REVIEW

File: MSAL/IdentityCore [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit dc9c4318c9f3d88fee2688655e75da798ddf3409
```
Comment (kaisong1990) [inline MSAL/IdentityCore:1]: This pull request does not update CHANGELOG.md.

Please consider if this change would be noticeable to a partner or user and either update CHANGELOG.md or resolve this conversation.

**Review by kaisong1990 — APPROVED** [review]
LGTM

File: MSAL/src/public/configuration/global/MSALLoggerConfig.h [hunk 2]
```diff
@@ -45,7 +59,14 @@ NS_ASSUME_NONNULL_BEGIN
...
  */
-@property (atomic) BOOL piiEnabled;
+@property (atomic) BOOL piiEnabled DEPRECATED_MSG_ATTRIBUTE("Use logMaskingLevel instead");
+
+/**
+ MSAL provides logging callbacks that assist in diagnostics. By default the library will not return any messages with any user or organizational information. However, this might make diagnosing issues difficult.
+ logMaskingLevel property can be used to adjust level of MSAL masking.
+ Default value is MSALLogMaskingSettingsMaskAllPII.
+*/
+@property (atomic) MSALLogMaskingLevel logMaskingLevel;
```
Comment (antrix1989) [inline MSAL/src/public/configuration/global/MSALLoggerConfig.h:69]: logMaskingLevel and piiEnabled are declared as "nonatomic" in IdentityCore.

**Review by antrix1989 — APPROVED** [review]

File: MSAL/src/public/configuration/global/MSALLoggerConfig.h (hunk 2 above)
Comment (oldalton) [reply MSAL/src/public/configuration/global/MSALLoggerConfig.h]: Good catch. Addressed. Thanks!

**Pull request merged**